### ISSUE TITLE
Tune socket buffer size(s)

### ISSUE THREAD

**Issue by jmpesp** [oxidecomputer/crucible#392]
`tokio::net::TcpSocket` offers both [set_recv_buffer_size](https://docs.rs/tokio/latest/tokio/net/struct.TcpSocket.html#method.set_recv_buffer_size) and [set_send_buffer_size](https://docs.rs/tokio/latest/tokio/net/struct.TcpSocket.html#method.set_send_buffer_size).

Tune these away from their defaults and investigate the impact on performance. This may also necessitate changing other things like the maximum number of in-flight jobs.

**Comment by leftwo** [comment]
Yeah, the flow control stuff between upstairs and downstairs I think will be effected directly by this.
